### PR TITLE
Document expected behavior for live environments with multiple app servers

### LINKED ISSUE #368: Article about multiple application servers on production environments
Story:
Article about multiple application servers on production environments so that users know how to effectively use SFTP to upload files on live

[Mar-10 3:45 PM] Alex Juchniewicz: @here can anyone else help me test this FTP use FileZilla? I am getting some strange errors on a client's live uploads folder
[Mar-10 3:45 PM] Alex Juchniewicz: using*
[Mar-10 3:45 PM] Suzanne "The Beast" Aldrich: what is the error message?
[Mar-10 3:46 PM] Alex Juchniewicz: 
        Status:      Starting upload of /Users/alexj/Pictures/bike.jpg
        Command: cd "/srv/bindings/3e85da4897c74faeb389aeb36d028623/files/2015"
        Error:        Directory /srv/bindings/3e85da4897c74faeb389aeb36d028623/files/2015: no such file or directory
        Command: cd "/srv/bindings/3e85da4897c74faeb389aeb36d028623/files"
        Error:        Directory /srv/bindings/3e85da4897c74faeb389aeb36d028623/files: no such file or directory
        Command: cd "/srv/bindings/3e85da4897c74faeb389aeb36d028623"
        Error:        Directory /srv/bindings/3e85da4897c74faeb389aeb36d028623: no such file or directory
        Command: cd "/srv/bindings"
        Response: New directory is: "/srv/bindings"
        Command: mkdir "3e85da4897c74faeb389aeb36d028623"
        Error:        mkdir /srv/bindings/3e85da4897c74faeb389aeb36d028623: permission denied
        Show more

[Mar-10 3:47 PM] Alex Juchniewicz: https://gist.github.com/anonymous/018b885e8adbd0b48da1
[Mar-10 3:47 PM] Suzanne "The Beast" Aldrich: Error:        Directory /srv/bindings/3e85da4897c74faeb389aeb36d028623/files/2015: no such file or directory
[Mar-10 3:47 PM] David Strauss: @AlexJuchniewicz Looks like a deleted binding
[Mar-10 3:47 PM] Suzanne "The Beast" Aldrich: no
[Mar-10 3:47 PM] David Strauss: Error:        Directory /srv/bindings/3e85da4897c74faeb389aeb36d028623: no such file or directory
[Mar-10 3:48 PM] Suzanne "The Beast" Aldrich: what is happening is the binding directory is in connection info
[Mar-10 3:48 PM] Suzanne "The Beast" Aldrich: you are reconnecting to a different appserver
[Mar-10 3:48 PM] David Strauss: Forget Valhalla. Not even the binding directory exists.
[Mar-10 3:48 PM] Suzanne "The Beast" Aldrich: so it has a different binding UUID
[Mar-10 3:48 PM] Alex Juchniewicz: its not bookmarked either
[Mar-10 3:48 PM] Alex Juchniewicz: client is seeing same issue
[Mar-10 3:48 PM] Suzanne "The Beast" Aldrich: yes, it is cached somewhere
[Mar-10 3:49 PM] Alex Juchniewicz: ok, then maybe somewhere on the back-end? because I have never connected to it before on my computer before just now
[Mar-10 3:49 PM] Suzanne "The Beast" Aldrich: https://www.evernote.com/shard/s42/sh/7ca4c215-2178-4052-b701-7dd97e45cecb/0987fa0d74b3bbc3663867c9f...
        Image

[image: https://www.evernote.com/shard/s42/sh/7ca4c215-2178-4052-b701-7dd97e45cecb/0987fa0d74b3bbc3663867c9f9c4a66b/deep/0/Site-Manager-and-FileZilla.png]
[Mar-10 3:49 PM] Suzanne "The Beast" Aldrich: make sure the advanced  default remote directory is blank
[Mar-10 3:50 PM] Alex Juchniewicz: yup
[Mar-10 3:50 PM] Suzanne "The Beast" Aldrich: https://www.evernote.com/shard/s42/sh/308857f7-eb90-410a-b370-77ad7e520707/dc3bb6d3da63d2cb85806b01e...
        Image

[image: https://www.evernote.com/shard/s42/sh/308857f7-eb90-410a-b370-77ad7e520707/dc3bb6d3da63d2cb85806b01e8426c57/deep/0/Banners-and-Alerts-and-Site-Manager-and-FileZilla.png]
[Mar-10 3:50 PM] Suzanne "The Beast" Aldrich: make sure to limite the number of simultaneous connectins to 1
[Mar-10 3:50 PM] Alex Juchniewicz: yup
[Mar-10 3:50 PM] Alex Juchniewicz: try adding yourself and see if you can - https://admin.dashboard.pantheon.io/sites/80c76a33-a29b-4c7e-94ec-5824f82f6c1f#live/deploys
[Mar-10 3:51 PM] Alex Juchniewicz: I added a non employee account and it still doesn't work
[Mar-10 3:52 PM] Nick (chewie) Stielau: @here atlas is back up.  MySQL ran out of memory.  We haven't seen that happen recently, but did have a few issues coming up from the RAX reboots.  We'll keep an eye on it.

[Mar-10 3:52 PM] Brian MacKinney: drush pauth success
[Mar-10 3:55 PM] Suzanne "The Beast" Aldrich: Status:      Starting upload of /Users/suzannealdrich/Pictures/Pantheon/pantheon-fist.png         Command: put "/Users/suzannealdrich/Pictures/Pantheon/pantheon-fist.png" "pantheon-fist.png"        Status:      local:/Users/suzannealdrich/Pictures/Pantheon/pantheon-fist.png => remote:/srv/bindings/3e85da4897c74faeb389aeb36d028623/files/2015/pantheon-fist.png      Status:      File transfer successful, transferred 12395 bytes in 1 second
[Mar-10 3:55 PM] Suzanne "The Beast" Aldrich: this time i connected to the appserver with the files mounted at that directory path
[Mar-10 3:55 PM] Alex Juchniewicz: hermm
[Mar-10 3:55 PM] Suzanne "The Beast" Aldrich: another time i connect might get the failover
[Mar-10 3:55 PM] Alex Juchniewicz: Should you have to do that though?
[Mar-10 3:55 PM] Suzanne "The Beast" Aldrich: do what?
[Mar-10 3:56 PM] Alex Juchniewicz: manually mount the path
[Mar-10 3:56 PM] Suzanne "The Beast" Aldrich: i didn't do that
[Mar-10 3:56 PM] Alex Juchniewicz: right right
[Mar-10 3:56 PM] Suzanne "The Beast" Aldrich: i have just logged in exactly as our documentation suggests
[Mar-10 3:56 PM] Alex Juchniewicz: read that wrong
[Mar-10 3:56 PM] Suzanne "The Beast" Aldrich: the host is appserver.live.80c76a33-a29b-4c7e-94ec-5824f82f6c1f.drush.in
[Mar-10 3:56 PM] Suzanne "The Beast" Aldrich: if you dig that host you get a couple of ips
[Mar-10 3:56 PM] Suzanne "The Beast" Aldrich: $ dig +short appserver.live.80c76a33-29b-4c7e-94ec-5824f82f6c1f.drush.in      23.253.170.178      104.130.220.65
[Mar-10 3:57 PM] Alex Juchniewicz: appserver.live.80c76a33-a29b-4c7e-94ec-5824f82f6c1f.drush.in
[Mar-10 3:57 PM] Alex Juchniewicz: I got the same appserver
[Mar-10 3:57 PM] Suzanne "The Beast" Aldrich: yes
[Mar-10 3:57 PM] Alex Juchniewicz: And mine is failing
[Mar-10 3:57 PM] Suzanne "The Beast" Aldrich: that is a name that can refer to two different endpionts
[Mar-10 3:57 PM] Alex Juchniewicz: right
[Mar-10 3:57 PM] Suzanne "The Beast" Aldrich: endpointf2e4f62a - 104.130.220.65:15335
[Mar-10 3:57 PM] Suzanne "The Beast" Aldrich: endpointc85730d4 - 23.253.170.178:21253
[Mar-10 3:57 PM] Alex Juchniewicz: So how is this fixed?
[Mar-10 3:58 PM] Suzanne "The Beast" Aldrich: you get to one with ssh 104.130.220.65.panth.io -t "cd /srv/bindings/3e85da4897c74faeb389aeb36d028623; bash --login"
[Mar-10 3:58 PM] Suzanne "The Beast" Aldrich: the other with ssh 23.253.170.178.panth.io -t "cd /srv/bindings/f4cf5bb013334b6ca40c2235f8696f85; bash --login"
[Mar-10 3:58 PM] Alex Juchniewicz: So you don't round robin all the time
[Mar-10 3:58 PM] Suzanne "The Beast" Aldrich: you can't prevent that, just make one connection at at ime
[Mar-10 3:58 PM] Suzanne "The Beast" Aldrich: do your uploads log out
[Mar-10 3:58 PM] Suzanne "The Beast" Aldrich: if they are keeping the same client open doing uploads this might happen i suppose
[Mar-10 3:58 PM] Suzanne "The Beast" Aldrich: reconnecting, the client might assume the same directory
[Mar-10 3:58 PM] Alex Juchniewicz: I think that's a little much requiring our users to have to do that, non tech people want to connect and get their file
[Mar-10 3:58 PM] Alex Juchniewicz: it should be stupid simple
[Mar-10 3:59 PM] Suzanne "The Beast" Aldrich: most of the time people are uploading files via the CMS
[Mar-10 3:59 PM] Suzanne "The Beast" Aldrich: not SFTP
[Mar-10 3:59 PM] Alex Juchniewicz: It's still an issue though
[Mar-10 3:59 PM] Suzanne "The Beast" Aldrich: our filesystem is designed for that
[Mar-10 3:59 PM] Suzanne "The Beast" Aldrich: take it up with engineering
[Mar-10 3:59 PM] Suzanne "The Beast" Aldrich: can't be everything for everybody
[Mar-10 3:59 PM] Alex Juchniewicz: I understand, I just see it as a problem, this is not the first ticket I have seen with this behavior.
[Mar-10 4:00 PM] Alex Juchniewicz: That's a simple thing lol
[Mar-10 4:00 PM] Suzanne "The Beast" Aldrich: it won't be the last either
[Mar-10 4:00 PM] Alex Juchniewicz: To be able to connect to SFTP and get your file?
[Mar-10 4:00 PM] Suzanne "The Beast" Aldrich: in return the sites don't go down when a million visitors hit
[Mar-10 4:00 PM] Suzanne "The Beast" Aldrich: use rsync if it's a problem
[Mar-10 4:00 PM] Suzanne "The Beast" Aldrich: or a different client then filezilla
[Mar-10 4:01 PM] Suzanne "The Beast" Aldrich: workarounds exist.
[Mar-10 4:01 PM] Alex Juchniewicz: All I am saying is that when you have tech people and that continues to come up as a problem, there should be a solution so they don't have to file tickets because it's something that OUR system causes to happen
[Mar-10 4:01 PM] Alex Juchniewicz: Moving on.
[Mar-10 4:04 PM] Brian MacKinney: Kalabox Google group: https://groups.google.com/a/kalamuna.com/d/forum/kalabox-friends
[Mar-10 4:04 PM] Suzanne "The Beast" Aldrich: it's not something our system is causing
[Mar-10 4:04 PM] Suzanne "The Beast" Aldrich: filezilla as an sftp client is making a lazy assumption
[Mar-10 4:05 PM] Suzanne "The Beast" Aldrich: that the client host will always resolve to the same server and the directory will be the same
[Mar-10 4:05 PM] Suzanne "The Beast" Aldrich: so, this works as designed
[Mar-10 4:06 PM] Alex Juchniewicz: ok :)
[Mar-10 4:08 PM] David Strauss: Suzanne also corrected me on a sloppy use of terminology. The binding isn't "deleted," and I shouldn't suggest that. It's either deleted or not on that server.
[Mar-10 4:09 PM] Suzanne "The Beast" Aldrich: If i am at the point of correcting @davidstrauss I'm either doing it right or about to get canned XD

### PR DESCRIPTION
**Pull request by rachelwhitton**
Closes #368 

New section **Multiple Application Containers** added to [All About Application Containers](https://pantheon.io/docs/articles/sites/all-about-application-containers/)

@atjuch and @ari-gold can you review?

### CODE REVIEW

**Comment by ari-gold** [comment]
+1

File: source/docs/articles/sites/all-about-application-containers.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Pantheon's infrastructure includes a number of layers. Our edge layer provides r
 
 Pantheon's infrastructure is based on a grid model. We serve our customers by provisioning isolated linux container with an optimized PHP stack in place. Each container includes its own Nginx, APC cache, and PHP worker agent. They are deployed with a checkout of your codebase and service-bindings to use a dedicated MySQL container, networked file filesystem, and optionally Redis object cache and Apache Solr search indexing.
 
-Every environment for your site (Dev, Test, Live) runs on its own container. In the case of a Live site, at the Pro level and above you can have many containers serving your site.
+Every environment for your site (Dev, Test, Live) runs on its own container. In the case of a Live site, at the Professional level and above you can have [multiple containers](#multiple-application-containers) serving your site.
```
Comment (bmackinney) [inline source/docs/articles/sites/all-about-application-containers.md]: Business and above. Pro sites have 1 container.

File: source/docs/articles/sites/all-about-application-containers.md [hunk 2]
```diff
@@ -42,3 +42,8 @@ While these are related topics, they need to be evaluated separately for your pr
 Typically, the best practice is to optimize for performance first, and then begin to look at handling scale. In most cases, if you are able to deliver an individual request quickly, handling more requests is primarily a matter of adding more containers up to the point where other bottlenecks (typically SQL queries) emerge. At that point, the next step really depends on your application.
 
 Because Pantheon does not handicap or limit free sandbox or dev instances, you should be able to get a good sense of your sites live performance as you develop. Support is happy to answer questions about site performance if you feel it's not up to where it should be. We also provide tools like [New Relic](/docs/articles/sites/newrelic/new-relic-performance-analysis) to give you insights about your site's performance.
+
+## Multiple Application Containers
```
Comment (bmackinney) [inline source/docs/articles/sites/all-about-application-containers.md]: same

Comment (rachelwhitton) [inline source/docs/articles/sites/all-about-application-containers.md]: Updated, thanks @bmackinney !

**Comment by atjuch** [comment]
+1

**Comment by bmackinney** [comment]
:+1:

**Pull request merged**